akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/LeveldbReadJournal.scala

```diff
@@ -22,9 +22,135 @@ import akka.util.ByteString
+ * efficiency reasons the query side retrieves the events in batches that sometimes can
+ * be delayed up to the configured `refresh-interval` or given [[akka.persistence.query.RefreshInterval]]
+ * hint.
+ *
+ * The stream is completed with failure if there is a failure in executing the query in the
+ * backend journal.
+ *
+ * == AllPersistenceIds ==
+ *
+ * [[akka.persistence.query.AllPersistenceIds]] is used for retrieving all `persistenceIds` of all
+ * persistent actors.
+ *
+ * The order of the returned event stream is unordered and you can expect different order
+ * for multiple executions of the query.
+ *
+ * The query supports two different completion modes:
+ * <ul>
+ * <li>The stream is not completed when it reaches the end of the currently used `persistenceIds`,
+ *     but it continues to push new `persistenceIds` when new persistent actors are created.
+ *     This is the default mode that is used when no hints are given. It can also be specified with
+ *     hint [[akka.persistence.query.RefreshInterval]].</li>
+ * <li>The stream is completed when it reaches the end of the currently used `persistenceIds`.
+ *     This mode is specified with hint [[akka.persistence.query.NoRefresh]].</li>
+ * </ul>
+ *
+ * The LevelDB write journal is notifying the query side as soon as new `persistenceIds` are
+ * created and there is no periodic polling or batching involved in this query.
+ *
+ * The stream is completed with failure if there is a failure in executing the query in the
+ * backend journal.
+ *
+ * == EventsByTag ==
+ *
+ * [[akka.persistence.query.EventsByTag]] is used for retrieving events that were marked with
+ * a given tag, e.g. all events of an Aggregate Root type.
+ *
+ * To tag events you create an [[akka.persistence.journal.EventAdapter]] that wraps the events
+ * in a [[akka.persistence.journal.leveldb.Tagged]] with the given `tags`.
+ *
+ * You can retrieve a subset of all events by specifying `offset`, or use `0L` to retrieve all
+ * events with a given tag. The `offset` corresponds to an ordered sequence number for
+ * the specific tag. Note that the corresponding offset of each event is provided in the
+ * [[akka.persistence.query.EventEnvelope]], which makes it possible to resume the
+ * stream at a later point from a given offset.
+ *
+ * In addition to the `offset` the `EventEnvelope` also provides `persistenceId` and `sequenceNr`
+ * for each event. The `sequenceNr` is the sequence number for the persistent actor with the
+ * `persistenceId` that persisted the event. The `persistenceId` + `sequenceNr` is an unique
+ * identifier for the event.
+ *
+ * The returned event stream is ordered by the offset (tag sequence number), which corresponds
+ * to the same order as the write journal stored the events. The same stream elements (in same order)
+ * are returned for multiple executions of the query. Deleted events are not deleted from the
+ * tagged event stream.
```

Comment:
yeah, I found it to be a nice feature (but someone probably wants it differently)